### ISSUE TITLE
🐞 runtime error: index out of range [0] with length 0

### ISSUE THREAD

**Issue by gerhard** [dagger/dagger#6272]
### What is the issue?

A module that used to work in Dagger `v0.9.3` doesn't work in `v0.9.4`:

`dagger-0.9.4 -m github.com/vito/daggerverse/test call testcontainers`

### Log output

```console
┃ Error: response from query: input:1: main.testcontainers failed to get function output directory: process "/runtime" did not complete successfully: exit code: 2
✘ testcontainers ERROR [3.01s]
✘ exec /runtime ERROR [3.01s]
┃ input:1: testcontainers.dockerService failed to get function output directory: process "/runtime" did not complete successfully: exit code: 2
✘ dockerService ERROR [2.42s]
✘ exec /runtime ERROR [2.41s]
┃ json: error calling MarshalJSON for type *main.Service: input:1: docker.daemon.service failed to get function output directory: process "/runtime" did not complete successfully
┃ exit code: 2
✘ service ERROR [1.42s]
✘ exec /runtime ERROR [1.42s]
┃ json: error calling MarshalJSON for type *main.Service: input:1: container.from.withEnvVariable.withMountedCache runtime error: index out of range [0] with length 0
• Engine: 46dc831a4e27 (version v0.9.4)
⧗ 7.25s ✔ 69 ∅ 49 ✘ 7
```

### Steps to reproduce

<img width="1432" alt="image" src="https://github.com/dagger/dagger/assets/3342/58c62b6a-d307-4039-84d0-b49c447ed5be">

### SDK version

CLI v0.9.4

### OS version

macOS 12.7.1

**Comment by jedevc** [comment]
cc @vito (we discussed this one in discord briefly)

PR incoming soon!

**Comment by vito** [comment]
@jedevc I think this one might be different, but it would be great if I was wrong. (index out of range vs. nil pointer dereference)

**Comment by jedevc** [comment]
:scream: yup, you're right this is different.

Thankfully... this is actually already fixed in main: https://github.com/dagger/dagger/pull/6257.

It's cause by weirdness where a core field defined on a module type will not get properly set to nil (but instead have a strange non-zero value that's not really valid).

However, the index error is very strange, we shouldn't have panics like this. I'll submit a fix to get a better message out of this.

**Comment by jedevc** [comment]
Ok, looks like there's another issue with `BuildOpts` from quartz's daggerverse: https://github.com/quartz-technology/daggerverse/pull/2 

Previously `BuildOpts` wasn't being code-gened (since it didn't have any functions): https://github.com/dagger/dagger/commit/77760c2754575695f6a9329c79c4c6a2d6078a65#diff-42df7f581f0c9e41a3320a7be6c79027f0b373d77e57231d767b3539643a4da8L155-R152

But with the above refactor, it is. The correct solution is to not code-gen this because there's no link from the top-level main type to it (opening a PR soon).

**Comment by jedevc** [comment]
Since this is already fixed in main with #6257, I'm going to close this.

The weirdness from `BuildOpts` will be resolved in #6282, or the upstream quartz-technology PR. The more general version of it is equivalent to https://github.com/dagger/dagger/issues/6186, where we can accidentally generate two different structs with the same name.

**Comment by vito** [comment]
> @vito also just a note - it looks like the panic stacktrace gets _completely_ swallowed up by our graqhql server. Not worth fixing now IMO, since [vito/dagql](https://github.com/vito/dagql/) is incoming 🎉 (also, I tried, and there's so many layers of panics, I have no idea where we even get the right stack trace from)

@jedevc Oh yeah the panic swallowing drives me nuts. Will keep an eye on that and make sure DagQL handles it better.
